src/components/Graph/RainbowChart/RainbowChart.tsx

```diff
@@ -0,0 +1,220 @@
+import { Stack, Text } from '@chakra-ui/react'
+import { useColorModeValue, useToken } from '@chakra-ui/system'
+import { curveLinear } from '@visx/curve'
+import { Group } from '@visx/group'
+import { ScaleSVG } from '@visx/responsive'
+import { Text as VisxText } from '@visx/text'
+import { AreaSeries, AreaStack, Axis, Margin, Tooltip, XYChart } from '@visx/xychart'
+import { extent, Numeric } from 'd3-array'
+import dayjs from 'dayjs'
+import omit from 'lodash/omit'
+import React, { useMemo } from 'react'
+import { useSelector } from 'react-redux'
+import { Amount } from 'components/Amount/Amount'
+import { AssetIcon } from 'components/AssetIcon'
+import { RainbowData } from 'hooks/useBalanceChartData/useBalanceChartData'
+import { useLocaleFormatter } from 'hooks/useLocaleFormatter/useLocaleFormatter'
+import { selectAssets, selectSelectedLocale } from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+import { colors } from 'theme/colors'
+
+export type RainbowChartProps = {
+  data: RainbowData[]
+  width: number
+  height: number
+  color: string
+  margin?: Margin
+}
+
+// https://codesandbox.io/s/github/airbnb/visx/tree/master/packages/visx-demo/src/sandboxes/visx-xychart?file=/customTheme.ts:50-280
+export const RainbowChart: React.FC<RainbowChartProps> = ({
+  data,
+  width,
+  height,
+  color,
+  margin = { top: 0, right: 0, bottom: 0, left: 0 },
+}) => {
+  const selectedLocale = useAppSelector(selectSelectedLocale)
+  const assetIds = useMemo(() => Object.keys(omit(data[0], ['date', 'total'])), [data])
+  const assets = useSelector(selectAssets)
+
+  const {
+    number: { toFiat },
+  } = useLocaleFormatter()
+  const magicXAxisOffset = 37
+
+  type Accessor = (d: RainbowData) => number
```

Comment:
Export outside of the component 🙏🏽

Reply:
curious about the rationale here if the usage is entirely scoped to this component, a view-layer concern?

Reply:
Just a style consistency OCD concern, since we usually do imports - types - constants - methods/components.

But the rationale of collocating it next to usage in this case is also valid (although not consistent)